backend/app/Http/Controllers/TransactionController.php

```diff
@@ -5,20 +5,21 @@
 use App\Http\Resources\TransactionResource;
 use App\Models\Account;
 use App\Models\Transaction;
+use Illuminate\Http\Request;
 
 class TransactionController extends Controller
 {
-    public function index(Account $account = null)
+    public function index(Request $request, Account $account = null)
     {
-        // For /accounts/{account_id}/transactions
-        if ($account) {
-            // For shallow nesting
-            // See https://laravel.com/docs/8.x/controllers#shallow-nesting
-            return TransactionResource::collection($account->accountTransactions);
+        //Get transactions by all or through accounts
+        $transactions = $account ? $account->accountTransactions : Transaction::all();
```

Comment:
reference and highlight of shallow nesting got removed

put again the reference for the shallow nesting, as well as highlight the portion for the shallow nesting

Reply:
Added comment and reference